### ISSUE TITLE
Cannot open a geordi shell or console

### ISSUE THREAD

**Issue by pmeinhardt** [makandra/geordi#49]
Hi there 👋 

I've been trying to open a shell (and console) on a recent project, but received this error message instead:

```text
$ geordi shell staging

# Opening a shell on staging
> Connecting to c23.bitcrowd-2.makandra.de
usage: ssh [-1246AaCfGgKkMNnqsTtVvXxYy] [-b bind_address] [-c cipher_spec]
           [-D [bind_address:]port] [-E log_file] [-e escape_char]
           [-F configfile] [-I pkcs11] [-i identity_file]
           [-J [user@]host[:port]] [-L address] [-l login_name] [-m mac_spec]
           [-O ctl_cmd] [-o option] [-p port] [-Q query_option] [-R address]
           [-S ctl_path] [-W host:port] [-w local_tun[:remote_tun]]
           [user@]hostname [command]

x Something went wrong.
```

It looks a bit like the host and other options are possibly not extracted correctly,
as the above ssh usage is also printed when running `ssh`, `ssh @` (i.e. with missing values).

Here's a gist with the relevant Capistrano deploy config (modified):
https://gist.github.com/pmeinhardt/2ca7a906fc82d6e0114d2d3932b0effe

If I can help with any additional information or a debug session, let me know.

Side note: I was wondering why geordi apparently uses regular expressions to extract the server and user information instead of either using Capistrano to load the configuration files or at least using a proper Ruby parser (which might avoid such issues?).

Thanks a lot 🙂

**Comment by codener** [comment]
Please see #44.

**Comment by pmeinhardt** [comment]
Argh, thanks a lot @codener for the prompt reply and for pointing out the other, related issue.
Do you have any plans to fix these issues (and use a more reliable way of extracting the config values)?

**Comment by codener** [comment]
We have no plans to do something about this, because this has never been an issue to us—we're using one-line server definitions.

As stated in #44, we'd surely accept a PR. Please note though, that it **must** support older versions of Capistrano. If I remember right, the changing config strategies of Capistrano were a major reason we returned to regular expressions, cf. https://github.com/makandra/geordi/commit/f6a950c24d3cb408d86000d00ffd75254542c572.

**Comment by pmeinhardt** [comment]
Thanks for this background information. So geordi aims to support all versions of Capistrano? 🤔 
To me it would seem natural to drop support for outdated Capistrano versions at some point.

**Comment by codener** [comment]
Well, as long as Geordi is being used on projects with Capistrano 2, support will have to remain.

**Comment by pmeinhardt** [comment]
Alright. That makes sense (it sounded a bit like "support all older capistrano versions" at first ). 👍 

I haven't looked into it in too much detail, but this could potentially be solved by having a specific "adapter" per Capistrano version to deal with the different approaches to configuration. I'll see if I can find some time to create a PR if possible.

**Comment by denzelem** [comment]
Fixed in #51 and included in release 1.10.0.
